module-services/service-desktop/endpoints/msc/MscEndpoint.cpp

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include <endpoints/msc/MscEndpoint.hpp>
+#include <endpoints/HttpEnums.hpp>
+#include <log/log.hpp>
+#include <module-sys/SystemManager/include/SystemManager/SystemManagerCommon.hpp>
+#include <endpoints/message/Sender.hpp>
+
+namespace sdesktop::endpoints
+{
+    auto MscEndpoint::handle(Context &context) -> void
+    {
+        switch (context.getMethod()) {
+        case http::Method::post: {
+            const auto &body = context.getBody();
+
+            auto msg = body["msg"].string_value();
```

Comment:
how about creating a key called "msc" holding bool value instead of checking if the key "msg" is in JSON and then if the message in it contains proper string just to check
if(body[json::msc].bool_value()){// code here}
And add "msc" to a variable in the proper namespace as it is done in other endpoints

Reply:
Another proposition to make it more general is rename this endpoint to RebootEndpint / shutdownEndpoint and create an enum with different reasons
and then check like this:
if(body[json::rebootReason].int_value() == ReabootReason::MSC){// code here}
else{ //TO DO 
}

Reply:
It will give us the possibility to implement reboot devices without touching them if we need it in the future